src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -201,21 +201,61 @@ public void importEntryWithDuplicateCheck(BibDatabaseContext bibDatabaseContext,
         Optional<BibEntry> existingDuplicateInLibrary = new DuplicateCheck(Globals.entryTypesManager).containsDuplicate(bibDatabaseContext.getDatabase(), entryToInsert, bibDatabaseContext.getMode());
         if (existingDuplicateInLibrary.isPresent()) {
             DuplicateResolverDialog dialog = new DuplicateResolverDialog(existingDuplicateInLibrary.get(), entryToInsert, DuplicateResolverDialog.DuplicateResolverType.IMPORT_CHECK, bibDatabaseContext, stateManager, dialogService, preferencesService);
-            switch (dialogService.showCustomDialogAndWait(dialog).orElse(DuplicateResolverDialog.DuplicateResolverResult.BREAK)) {
+
+            DuplicateResolverDialog.DuplicateResolverResult dialogResult;
+
+            int allEntriesCase = preferencesService.getEntryResolverResult();
+            int entriesQuantity = preferencesService.getEntryPasteQuantity();
+
+            if (preferencesService.getGuiPreferences().mergeApplyToAllEntriesProperty().get()) {
+                 switch (allEntriesCase) {
```

Comment:
A better idea would be to store the string value in the preferences and parse it back to the enum value using valueOf.  Makes your code much simpler.

You can look at the GroupViewMode implementation in the prefs, eg..
https://github.com/JabRef/jabref/blob/1cb4e160af80bf7af9339c4cf8d90f388bfa3de3/src/main/java/org/jabref/preferences/JabRefPreferences.java#L1406-L1411